### PR TITLE
#213 Implemented validation of mandatory fields for transcriptions

### PR DESCRIPTION
**Pull request by rgraf**
The fields: language, value and edmRights are mandatory, and "source" becomes mandatory as soon as  "scope" field is in the target. Exception is thrown if mandatory fields are missing.

### CODE REVIEW

File: annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java [hunk 1]
```diff
@@ -845,6 +896,7 @@ public void validateWebAnnotation(Annotation webAnnotation)
 	    validateTag(webAnnotation);
 	    break;
 	case TRANSCRIBING:
+	    validateTranscribing(webAnnotation);
 	    validateEdmRights(webAnnotation);
```
Comment (gsergiu) [inline annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java]: validation of edmRights must be included in the vaidateTranscription method.

File: annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java [hunk 2]
```diff
@@ -646,6 +648,23 @@ private void validateTag(Annotation webAnnotation) throws ParamValidationExcepti
 	}
     }
 
+    /**
+     * Validation of transcribing.
+     * 
+     * @param webAnnotation
+     */
+    private void validateTranscribing(Annotation webAnnotation) throws ParamValidationException {
```
Comment (gsergiu) [inline annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java]: rename to validateTranscription

File: annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java [hunk 3]
```diff
@@ -646,6 +648,23 @@ private void validateTag(Annotation webAnnotation) throws ParamValidationExcepti
...
+		Body body = webAnnotation.getBody();
+		Target target = webAnnotation.getTarget();
+	
+		// the body type shouldn't be null at this stage
+		if (!(body instanceof SpecificResourceBody)) {
+		    throw new ParamValidationException(ParamValidationException.MESSAGE_WRONG_CLASS,
```
Comment (gsergiu) [inline annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java]: move this validation into the validateTranscribingWithSpecificResource method

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/transcribe/BaseTranscribingTest.java [hunk 4]
```diff
@@ -10,6 +10,7 @@
 
 	public static final String TRANSCRIPTION_MINIMAL = "/transcription/minimal.json";
```
Comment (gsergiu) [inline annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/transcribe/BaseTranscribingTest.java]: must update the minimal template with the mandatory fields.
The json templates for the create annotation method must be updated after making the fields mandatory.
The jsp template must indicate the mandatory fields

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/transcribe/BaseTranscribingTest.java [hunk 5]
```diff
@@ -10,6 +10,7 @@
...
 	public static final String TRANSCRIPTION_WITH_RIGHTS = "/transcription/transcription-with-rights.json";
+	public static final String TRANSCRIPTION_WITHOUT_RIGHTS = "/transcription/transcription-without-rights.json";
```
Comment (gsergiu) [inline annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/transcribe/BaseTranscribingTest.java]: I don't see this file used, a unit test to check the the validation exception must be written

File: annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java (hunk 1 above)
Comment (rgraf) [reply annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java]: Validation method for edmRights moved to validateTranscription

File: annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java (hunk 2 above)
Comment (rgraf) [reply annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java]: Renamed to validateTranscription

File: annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java (hunk 3 above)
Comment (rgraf) [reply annotation-web/src/main/java/eu/europeana/annotation/web/service/impl/AnnotationServiceImpl.java]: moved into validateTranscriptionWithSpecificResource - also renamed this method similar like above

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/transcribe/BaseTranscribingTest.java (hunk 4 above)
Comment (rgraf) [reply annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/transcribe/BaseTranscribingTest.java]: Added mandatory fields edmRights and value to minimal json template. JSP template documentation extended with description of mandatory fields.

File: annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/transcribe/BaseTranscribingTest.java (hunk 5 above)
Comment (rgraf) [reply annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/transcribe/BaseTranscribingTest.java]: I prepared this test, and see that server returns as expected ParamValidationException that extends HttpException. But Exception test fails for unclear reason with message: org.opentest4j.AssertionFailedError: Unexpected exception type thrown ==> expected: <org.apache.commons.httpclient.HttpException> but was: <org.opentest4j.AssertionFailedError> even when returned exception is correct.

Comment (gsergiu) [reply annotation-client/src/test/java/eu/europeana/annotation/client/integration/webanno/transcribe/BaseTranscribingTest.java]: at the client side we don't have the Http Exception, it is enough to verify sta status code

**Pull request merged**